### PR TITLE
cppcheck and coverity fixes

### PR DESCRIPTION
**Pull request by heckflosse**
This is for collecting cppcheck and coverity fixes for rt 5.8.
Pr created to get LGTM analysis

### CODE REVIEW

File: rtengine/profilestore.cc [hunk 1]
```diff
@@ -261,10 +261,10 @@ bool ProfileStore::parseDir (Glib::ustring& realPath, Glib::ustring& virtualPath
     return fileFound;
 }
 
-int ProfileStore::findFolderId (const Glib::ustring &path)
+int ProfileStore::findFolderId (const Glib::ustring &path) const
 {
     // initialization must have been done when calling this
-    for (std::vector<Glib::ustring>::iterator i = folders.begin(); i != folders.end(); ++i) {
+    for (std::vector<Glib::ustring>::const_iterator i = folders.begin(); i != folders.end(); ++i) {
```
Comment (Floessie) [inline rtengine/profilestore.cc]: One could use a simple index here, as it is returned later.

File: rtgui/profilestorecombobox.cc [hunk 2]
```diff
@@ -145,18 +144,16 @@ void ProfileStoreComboBox::updateProfileList ()
...
 
-Gtk::TreeIter ProfileStoreComboBox::findRowFromEntry_ (Gtk::TreeModel::Children childs, const ProfileStoreEntry *pse)
+Gtk::TreeIter ProfileStoreComboBox::findRowFromEntry_ (Gtk::TreeModel::Children childs, const ProfileStoreEntry *pse) const
 {
-    Gtk::TreeModel::Row row;
-    Gtk::TreeIter rowInSubLevel;
 
-    for (Gtk::TreeModel::Children::iterator iter = childs.begin(); iter != childs.end(); ++iter) {
-        row = *iter;
+    for (const auto iter : childs) {
```
Comment (Floessie) [inline rtgui/profilestorecombobox.cc]: Did you mean `const auto& iter`?

**Review by Floessie — COMMENTED** [review]
Very nice, Ingo! :+1:

File: rtengine/profilestore.cc (hunk 1 above)
Comment (heckflosse) [reply rtengine/profilestore.cc]: Fixed.

File: rtgui/profilestorecombobox.cc (hunk 2 above)
Comment (heckflosse) [reply rtgui/profilestorecombobox.cc]: Fixed.

**Comment by heckflosse** [comment]
Any objections to get this stuff into RT 5.7? @Floessie @Beep6581 @Hombre57 @Thanatomanic @Desmis

**Comment by Hombre57** [comment]
@heckflosse No objection from me.

**Comment by Floessie** [comment]
@heckflosse Nor from me. :+1:

**Comment by heckflosse** [comment]
So we are three. Merging... :smiley:

**Comment by Desmis** [comment]
@heckflosse 
No problem for me :)

**Pull request merged**